lib/Dialect/CGGI/IR/CGGIOps.td

```diff
@@ -57,6 +59,17 @@ def CGGI_NotOp : CGGI_Op<"not", [
   let summary = "Logical NOT of two ciphertexts";
 }
 
+def CGGI_PackedOp : CGGI_Op<"packed", [
+    Pure]> {
```

Comment:
This is also a Commutative operation, right? I also think you could use `SameOperandsAndResultType` - even if you have an attribute, i think that the type matching only occurs on values (I found an example, at least: https://github.com/google/jax/blob/b1066ee41373991bb66c5a56b2874dc5455b7f70/jaxlib/mosaic/dialect/tpu/tpu.td#L237)